File: org/cooklang/Ingredient.java
package org.cooklang;

public class Ingredient implements DirectionItem {
    public String name;

    private Float quantityFloat;
    private String quantityString;
    public String units;

    public void setName(String newName) {
        this.name = newName;
    }

    public void setQuantityFloat(Float newQuantity) {
        this.quantityFloat = newQuantity;
    }

    public void setQuantityString(String newQuantity) {
        this.quantityString = newQuantity;
    }

    public void setUnits(String newUnits) {
        this.units = newUnits;
    }

    public String toString() {
        return String.format("%s %s %s %s", name, quantityFloat, quantityString, units);
    }
}

File: parser/src/main/java/org/cooklang/parser/Ingredient.kt
package org.cooklang.parser

import org.cooklang.parser.DirectionItem

data class Ingredient(
    var name: String? = null,
    var quantityFloat: Float? = null,
    var quantityString: String? = null,
    var units: String? = null //maybe should be enum?
) : DirectionItem{
    override fun getDisplayString(): String {
        return "$name"
    }
}